hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/PipelineSyncTask.java

```diff
@@ -64,17 +69,16 @@ public PipelineSyncTask(ReconPipelineManager pipelineManager,
   }
 
   @Override
-  protected synchronized void run() {
+  public void run() {
     try {
       while (canRun()) {
-        long start = Time.monotonicNow();
-        List<Pipeline> pipelinesFromScm = scmClient.getPipelines();
-        reconPipelineManager.initializePipelines(pipelinesFromScm);
-        syncOperationalStateOnDeadNodes();
-        LOG.info("Pipeline sync Thread took {} milliseconds.",
-            Time.monotonicNow() - start);
-        recordSingleRunCompletion();
-        wait(interval);
+        try {
+          lock.writeLock().lock();
```

Comment:
We can remove taking lock here as lock is already taken inside for complete method. Plz handle for other place

Reply:
@sumitagrawl - changes fixed and pushed, pls re-review.